src/providers/hana/qgshanaprovider.cpp

```diff
@@ -0,0 +1,1780 @@
+  {
+    conn->execute( sql );
+    conn->commit();
+  }
+  catch ( const QgsHanaException &ex )
+  {
+    pushError( tr( "HANA error while deleting attributes: %1" )
+               .arg( QgsHanaUtils::formatErrorMessage( ex.what(), false ) ) );
+    conn->rollback();
+    return false;
+  }
+
+  readAttributeFields();
+
+  return true;
+}
+
+bool QgsHanaProvider::renameAttributes( const QgsFieldNameMap &fieldMap )
+{
+  if ( mIsQuery )
+    return false;
+
+  QgsHanaConnectionRef conn( mUri );
+  if ( conn.isNull() )
+    return false;
+
+  try
+  {
+    for ( QgsFieldNameMap::const_iterator it = fieldMap.begin(); it != fieldMap.end(); ++it )
+    {
+      int fieldIndex = it.key();
+      if ( fieldIndex < 0 || fieldIndex >= mAttributeFields.count() )
+      {
+        pushError( tr( "Invalid attribute index: %1" ).arg( fieldIndex ) );
+        return false;
+      }
+
+      if ( mAttributeFields.indexFromName( it.value() ) >= 0 )
+      {
+        pushError( tr( "Error renaming field %1: name '%2' already exists" ).arg( fieldIndex ).arg( it.value() ) );
+        return false;
+      }
+
+      QString sql = QStringLiteral( "RENAME COLUMN %1.%2.%3 TO %4" ).arg(
+                      QgsHanaUtils::quotedIdentifier( mSchemaName ), QgsHanaUtils::quotedIdentifier( mTableName ),
+                      QgsHanaUtils::quotedIdentifier( mAttributeFields.at( fieldIndex ).name() ),
+                      QgsHanaUtils::quotedIdentifier( it.value() ) );
+      conn->execute( sql );
+    }
+
+    conn->commit();
+  }
+  catch ( const QgsHanaException &ex )
+  {
+    pushError( tr( "HANA error while renaming attributes: %1" )
+               .arg( QgsHanaUtils::formatErrorMessage( ex.what(), false ) ) );
+    conn->rollback();
+    return false;
+  }
+
+  readAttributeFields();
+
+  return true;
+}
+
+bool QgsHanaProvider::changeGeometryValues( const QgsGeometryMap &geometryMap )
+{
+  if ( geometryMap.isEmpty() )
+    return true;
+
+  if ( mIsQuery )
+    return false;
+
+  if ( mGeometryColumn.isEmpty() )
+    return false;
+
+  QgsHanaConnectionRef conn( mUri );
+  if ( conn.isNull() )
+    return false;
+
+  QString sql = QStringLiteral( "UPDATE %1.%2 SET %3 = ST_GeomFromWKB(?, %4) WHERE %5 = ?" ).arg(
+                  QgsHanaUtils::quotedIdentifier( mSchemaName ), QgsHanaUtils::quotedIdentifier( mTableName ),
+                  QgsHanaUtils::quotedIdentifier( mGeometryColumn ), QString::number( mSrid ),
+                  QgsHanaUtils::quotedIdentifier( mFidColumn ) );
+
+  try
+  {
+    PreparedStatementRef stmt = conn->prepareStatement( sql );
+
+    for ( QgsGeometryMap::const_iterator it = geometryMap.begin(); it != geometryMap.end(); ++it )
+    {
+      QgsFeatureId fid = it.key();
+      // skip added features
+      if ( FID_IS_NEW( fid ) )
+        continue;
+
+      QByteArray wkb = it->asWkb();
+      stmt->setBinary( 1, makeNullable<vector<char>>( wkb.begin(), wkb.end() ) );
+      stmt->setLong( 2, fid );
+      stmt->executeUpdate();
```

Comment:
It can happen in situations like when you run an in-place processing algorithm to buffer every polygon by 10m or similar.
For this case, batch updating would make a huge difference in terms of additional network roundtrips avoided.